android/build.gradle

```diff
@@ -1,102 +1,55 @@
-            }
-        }
+// Top-level build file where you can add configuration options common to all sub-projects/modules.
 
-        dependencies {
-            classpath('com.android.tools.build:gradle:4.1.3')
-            classpath "org.jetbrains.kotlin:kotlin-gradle-plugin:1.6.10"
-            classpath 'org.jlleitschuh.gradle:ktlint-gradle:10.0.0'
+buildscript {
+    ext {
+        buildToolsVersion = "31.0.0"
+        minSdkVersion = 23
+        compileSdkVersion = 31
+        targetSdkVersion = 31
+
+        if (System.properties['os.arch'] == "aarch64") {
+            // For M1 Users we need to use the NDK 24 which added support for aarch64
+            ndkVersion = "24.0.8215888"
+        } else {
+            // Otherwise we default to the side-by-side NDK version from AGP.
+            ndkVersion = "21.4.7075529"
         }
-}
-
-apply plugin: 'com.android.library'
-apply plugin: 'kotlin-android'
-apply plugin: "org.jlleitschuh.gradle.ktlint"
-
-def safeExtGet(prop, fallback) {
-    rootProject.ext.has(prop) ? rootProject.ext.get(prop) : fallback
-}
-
-android {
-    compileSdkVersion safeExtGet('compileSdkVersion', 31)
-    buildToolsVersion safeExtGet('buildToolsVersion', '31.0.0')
-
-    compileOptions {
-        sourceCompatibility JavaVersion.VERSION_1_8
-        targetCompatibility JavaVersion.VERSION_1_8
-    }
-
-    defaultConfig {
-        minSdkVersion safeExtGet('minSdkVersion', 23)
-        targetSdkVersion safeExtGet('targetSdkVersion', 31)
-
-        testInstrumentationRunner "androidx.test.runner.AndroidJUnitRunner"
-
-        multiDexEnabled true
     }
-
-    lintOptions {
-        abortOnError false
-    }
-
-    testOptions {
-        unitTests.returnDefaultValues = true
-    }
-}
-
-repositories {
-    maven {
-        // All of React Native (JS, Obj-C sources, Android binaries) is installed from npm
-        url "$rootDir/../node_modules/react-native/android"
+    repositories {
+        google()
+        mavenCentral()
     }
-    mavenCentral()
-    google()
-}
-
-dependencies {
-    implementation 'com.facebook.react:react-native:+'  // From node_modules
-    implementation "androidx.core:core-ktx:1.3.2"
-    implementation "org.jetbrains.kotlin:kotlin-stdlib-jdk7:1.6.10"
-
-    // Required -- JUnit 4 framework
-    testImplementation 'junit:junit:4.12'
-    // Optional -- Mockito framework
-    testImplementation 'org.mockito:mockito-core:1.10.19'
-    // Test final classes
-    testImplementation 'org.mockito:mockito-inline:2.13.0'
-    // helpers for kotlin
-    testImplementation "org.mockito.kotlin:mockito-kotlin:3.1.0"
-    // mock Android stuff
-    testImplementation "io.mockk:mockk:1.10.6"
-
-    // Onegini SDK
-    api('com.onegini.mobile.sdk.android:onegini-sdk:11.7.0@aar') {
-        transitive = true
+    dependencies {
+        classpath("com.android.tools.build:gradle:7.1.1")
+        classpath("com.facebook.react:react-native-gradle-plugin")
```

Comment:
I think the version of the plugin should be specified to avoid any autupdates that could break the build

Reply:
This dependency is actually contained in the node_modules(the place where the javascript package manager puts all it's dependencies). So that javascript package manager sets the version for this dependency.